src/main/java/filter/expression/Qualifier.java

```diff
@@ -659,19 +659,7 @@ public static boolean hasUpdatedQualifier(FilterExpression expr) {
                 } else if (!bMilestone.isPresent()) {
                     return -1;
                 } else {
-                    Optional<LocalDate> aDueDate = aMilestone.get().getDueDate();
-                    Optional<LocalDate> bDueDate = bMilestone.get().getDueDate();
-
-                    if (!aDueDate.isPresent() && !bDueDate.isPresent()) {
-                        return 0;
-                    } else if (!aDueDate.isPresent()) {
-                        return 1;
-                    } else if (!bDueDate.isPresent()) {
-                        return -1;
-                    } else {
-                        return -(TurboMilestone.getDueDateComparator()
-                            .compare(aMilestone.get(), bMilestone.get()));
-                    }
+                    return -(TurboMilestone.getDueDateComparator().compare(aMilestone.get(), bMilestone.get()));
```

Comment:
Try to revert the tests to its previous state (supposedly broken state) and
push. I think when it was supposedly broken, Travis passed (and the tests
are not under unstable).
On Mar 16, 2016 11:40 AM, "Muhammad Rais Fathin Mudzakir" <
notifications@github.com> wrote:

> In src/main/java/filter/expression/Qualifier.java
> <https://github.com/HubTurbo/HubTurbo/pull/1344#discussion_r56278337>:
>
> > @@ -659,19 +659,7 @@ public static boolean hasUpdatedQualifier(FilterExpression expr) {
> >                  } else if (!bMilestone.isPresent()) {
> >                      return -1;
> >                  } else {
> > -                    Optional<LocalDate> aDueDate = aMilestone.get().getDueDate();
> > -                    Optional<LocalDate> bDueDate = bMilestone.get().getDueDate();
> > -
> > -                    if (!aDueDate.isPresent() && !bDueDate.isPresent()) {
> > -                        return 0;
> > -                    } else if (!aDueDate.isPresent()) {
> > -                        return 1;
> > -                    } else if (!bDueDate.isPresent()) {
> > -                        return -1;
> > -                    } else {
> > -                        return -(TurboMilestone.getDueDateComparator()
> > -                            .compare(aMilestone.get(), bMilestone.get()));
> > -                    }
> > +                    return -(TurboMilestone.getDueDateComparator().compare(aMilestone.get(), bMilestone.get()));
>
> Hmm, I think the most possible reason it didn't fail was because I was
> testing on the wrong branch (was testing on master?). After the first
> comment, I tested it once again (around an hour after) and realised that
> there were 2 failing tests.
>
> The test should be fine since it's pretty simple, and the comparator also
> satisfies all the requirements
> <https://docs.oracle.com/javase/8/docs/api/java/util/Comparator.html>. I
> can't see any other reason why the test could pass the first time only.
>
> Maybe ask @HansNewbie <https://github.com/HansNewbie> for opinion?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/HubTurbo/HubTurbo/pull/1344/files/fbbda207b1d8ad02f4570370b9514d984ccf8100#r56278337>
>